tests/thread/host/host.cpp

```diff
@@ -241,53 +246,48 @@ void* tcs_thread(oe_enclave_t* enclave, size_t tcs_req_count)
 //   - successful ecalls increment a counter (tcs_used_thread_count) and wait
 //     for the total thread count to reach the test count (tcs_req_count)
 //   - unsuccessful ecalls increment a counter (tcs_out_thread_count)
-// tcs_used_thread_count is maintained in the enclave
-// tcs_out_thread_count is maintained in the host
 void test_tcs_exhaustion(oe_enclave_t* enclave)
 {
     std::vector<std::thread> threads;
     // Set the test_tcs_count to a value greater than the enclave TCSCount
     const size_t test_tcs_req_count = enclave->num_bindings * 2;
+    const size_t expected_out_of_threads = enclave->num_bindings;
     printf(
         "test_tcs_exhaustion - Number of TCS bindings in enclave=%zu\n",
         enclave->num_bindings);
 
     for (size_t i = 0; i < test_tcs_req_count; i++)
     {
-        threads.push_back(std::thread(tcs_thread, enclave, test_tcs_req_count));
+        threads.push_back(
+            std::thread(tcs_thread, enclave, expected_out_of_threads));
     }
 
     for (size_t i = 0; i < test_tcs_req_count; i++)
     {
         threads[i].join();
     }
 
-    // the local tcs_used_thread_count is a local copy of a value maintained on
-    // the enclave
     size_t tcs_used_thread_count = 0;
-    // enc_tcs_used_thread_count retrieves the value from the enclave
     OE_TEST(
         enc_tcs_used_thread_count(enclave, &tcs_used_thread_count) == OE_OK);
 
     printf(
         "test_tcs_exhaustion: tcs_count=%zu; num_threads=%zu; "
-        "num_out_threads=%zu\n",
-        test_tcs_req_count,
+        "num_out_of_tcs=%zu\n",
         tcs_used_thread_count,
+        test_tcs_req_count,
         g_tcs_out_thread_count.load());
 
-    // Crux of the test is to get OE_OUT_OF_THREADS i.e. to exhaust the TCSes
-    OE_TEST(g_tcs_out_thread_count > 0);
-    // Verifying that everything adds up fine
-    OE_TEST(
-        tcs_used_thread_count + g_tcs_out_thread_count == test_tcs_req_count);
-    // Sanity test that we are not reusing the bindings
+    // Assert that expected thread exhaustion failures have been reached.
+    OE_TEST(g_tcs_out_thread_count == expected_out_of_threads);
```

Comment:
This test is redundant since the test will just hang and not call `g_tcs_cv.notify_all()` if this condition were not already true by this point.

Reply:
Redundancy is fine. I feel having this condition makes it clear what the test expectations are.